src/tests/ftest/deployment/network_failure.py

```diff
@@ -112,7 +113,37 @@ def create_ip_to_host(self):
         results = run_pcmd(hosts=self.hostlist_servers, command=command)
         self.log.info("hostname -i results = %s", results)
 
-        return {result["stdout"][0]: NodeSet(str(result["hosts"])) for result in results}
+        ip_to_host = {}
+        for result in results:
+            ips_str = result["stdout"][0]
+            # There may be multiple IP addresses for one host.
+            ip_addresses = ips_str.split()
+            for ip_address in ip_addresses:
+                ip_to_host[ip_address] = NodeSet(str(result["hosts"]))
+
+        return ip_to_host
+
+    def create_host_to_ranks(self, ip_to_host, system_query_members):
+        """Create a dictionary of hostname to ranks.
+
+        Args:
+            ip_to_host (dict): create_ip_to_host output.
+            system_query_members (dict): Contents of dmg system query accessed with
+                ["response"]["members"].
+
+        Returns:
+            dict: Hostname to ranks mapping. Hostname (key) is string. Ranks (value) is
+                list of int.
+
+        """
+        host_to_ranks = defaultdict(list)
+        for member in system_query_members:
+            ip_addr = member["addr"].split(":")[0]
```

Comment:
Wouldn't `member` be the keys in the dictionary, not the values? Or maybe it's not a dictionary?

Reply:
dmg system query returns the following output.

```
{
  "response": {
    "members": [
      {
        "addr": "10.8.1.110:10001",
        "state": "joined",
        "fault_domain": "/wolf-110.wolf.hpdd.intel.com",
        "rank": 0,
        "incarnation": 851553937757569024,
        "uuid": "45b814df-7994-4be2-b8d6-89e106885ce2",
        "fabric_uri": "ofi+verbs;ofi_rxm://192.168.100.110:31317",
        "fabric_contexts": 19,
        "info": "",
        "last_update": "2022-09-08T23:55:24.330470073Z"
      },
```

`system_query_members` is the list of dictionaries in the `members` list.